connect-examples/oauth/node/index.js

```diff
@@ -106,64 +115,82 @@ app.get("/request_token", (req, res) => {
-        res.send(messages.displayStateError());
-    }
-
-    else if (req.query['error']) {
-        // Check to see if the seller clicked the Deny button and handle it as a special case.
-        if(("access_denied" === req.query['error']) && ("user_denied" === req.query["error_description"])) {
-            res.send(messages.displayError("Authorization denied", "You chose to deny access to the app."));
-        }
-        // Display the error and description for all other errors.
-        else {
-            res.send(messages.displayError(req.query["error"], req.query["error_description"]));
-        }
-    }
-    // When the response_type is "code", the seller clicked Allow
-    // and the authorization page returned the auth tokens.
-    else if ("code" === req.query["response_type"]) {
-        // Extract the returned authorization code from the URL
-        var { code } = req.query;
-
-        try {
-            let { result } = await oauthInstance.obtainToken({
-                // Provide the code in a request to the Obtain Token endpoint
-                code,
-                clientId: process.env.SQ_APPLICATION_ID,
-                clientSecret: process.env.SQ_APPLICATION_SECRET,
-                grantType: 'authorization_code'
-            });
-
-            let {
-                // Extract the returned access token from the ObtainTokenResponse object
-                accessToken,
-                refreshToken,
-                expiresAt,
-                merchantId
-            } = result;
-
-            // Because we want to keep things simple and we're using Sandbox,
-            // we call a function that writes the tokens to the page so we can easily copy and use them directly.
-            // In production, you should never write tokens to the page. You should encrypt the tokens and handle them securely.
-            res.send(messages.writeTokensOnSuccess(accessToken, refreshToken, expiresAt, merchantId));
-        } catch (error) {
-            // The response from the Obtain Token endpoint did not include an access token. Something went wrong.
-            if (error instanceof ApiError) {
-                res.send(messages.displayError('Exception', JSON.stringify(error.result)));
-            } else {
-                res.send(messages.displayError('Exception', JSON.stringify(error)));
-            }
-        }
+  console.log(req.query);
+  // Verify the state to protect against cross-site request forgery.
+  if (req.cookies["Auth_State"] !== req.query['state']) {
+    content = messages.displayStateError();
+    res.render('base', {
+      content: content
+    });
+  }
+
+  else if (req.query['error']) {
+    // Check to see if the seller clicked the Deny button and handle it as a special case.
+    if (("access_denied" === req.query['error']) && ("user_denied" === req.query["error_description"])) {
+      res.render(messages.displayError("Authorization denied", "You chose to deny access to the app."));
```

Comment:
I'm confused about this path. When does the seller get to the Accept/Deny permissions screen? I wasn't able to reach it when running locally with new accounts.